src/main/java/org/fungover/breeze/Matrix/Matrix3x3.java

```diff
@@ -0,0 +1,157 @@
+        if (values.length != 3 || values[0].length != 3) {
+            throw new IllegalArgumentException("Matrix must be 3x3.");
+        }
+        this.matrix = values;
+    }
+
+    public static Matrix3x3 identity3x3() {
+        return new Matrix3x3(new float[][]{
+                {1, 0, 0},
+                {0, 1, 0},
+                {0, 0, 1}
+        });
+    }
+
+//     Matrix multiplication
+    public Matrix3x3 multiply3x3(Matrix3x3 other) {
+        float[][] result = new float[3][3];
+        for (int i = 0; i < 3; i++) {
+            for (int j = 0; j < 3; j++) {
+                float sum = 0;
+                for (int k = 0; k < 3; k++) {
+                    sum += matrix[i][k] * other.matrix[k][j];
+                }
+                result[i][j] = sum;
+            }
+        }
+        return new Matrix3x3(result);
+    }
+
+    // Matrix transpose
+    public Matrix3x3 transpose3x3() {
+        float[][] transposed = new float[3][3];
+        for (int i = 0; i < 3; i++) {
+            for (int j = 0; j < 3; j++) {
+                transposed[j][i] = matrix[i][j];
+            }
+        }
+        return new Matrix3x3(transposed);
+    }
+
+    // Determinant of the matrix
+   public float determinant3x3() {
+        return matrix[0][0] * (matrix[1][1] * matrix[2][2] - matrix[1][2] * matrix[2][1])
+                - matrix[0][1] * (matrix[1][0] * matrix[2][2] - matrix[1][2] * matrix[2][0])
+                + matrix[0][2] * (matrix[1][0] * matrix[2][1] - matrix[1][1] * matrix[2][0]);
+
+    }
+    // Inverse the matrix
+    public Matrix3x3 inverse() {
+        float det = determinant3x3();
+        if (det == 0) {
+            throw new ArithmeticException("Matrix is not invertible.");
+        }
+
+        // Cofactor matrix calculation with correct signs
+        float[][] cofactorMatrix = new float[3][3];
+
+        // Calculate each cofactor with correct signs (checkered pattern of signs)
+        cofactorMatrix[0][0] = +(matrix[1][1] * matrix[2][2] - matrix[1][2] * matrix[2][1]);
+        cofactorMatrix[0][1] = -(matrix[1][0] * matrix[2][2] - matrix[1][2] * matrix[2][0]);
+        cofactorMatrix[0][2] = +(matrix[1][0] * matrix[2][1] - matrix[1][1] * matrix[2][0]);
+
+        cofactorMatrix[1][0] = -(matrix[0][1] * matrix[2][2] - matrix[0][2] * matrix[2][1]);
+        cofactorMatrix[1][1] = +(matrix[0][0] * matrix[2][2] - matrix[0][2] * matrix[2][0]);
+        cofactorMatrix[1][2] = -(matrix[0][0] * matrix[1][2] - matrix[0][2] * matrix[1][0]);
+
+        cofactorMatrix[2][0] = +(matrix[0][1] * matrix[1][2] - matrix[0][2] * matrix[1][1]);
+        cofactorMatrix[2][1] = -(matrix[0][0] * matrix[1][2] - matrix[0][2] * matrix[1][0]);
+        cofactorMatrix[2][2] = +(matrix[0][0] * matrix[1][1] - matrix[0][1] * matrix[1][0]);
+
+        // Print Cofactor Matrix
+        System.out.println("Cofactor Matrix:");
+        for (float[] row : cofactorMatrix) {
+            System.out.println(Arrays.toString(row));
+        }
```

Comment:
We don't want any printing in our library methods. Might be usefull for debugging and in that case we can maybe wrap it in an if statement that checks on some variable like.

private static final boolean DEBUG_MODE = false;

if( DEBUG_MODE) {
...
}

The Java compiler (javac) employs a feature called constant folding. When you declare a private static final boolean variable and set it to a constant value, the compiler treats it as a compile-time constant. If the variable is used in a conditional statement like if (DEBUG_MODE) { ... }, and the condition is always false, the compiler will optimize the code by removing the entire block inside the if statement during compilation.